server/models/availability.js

```diff
@@ -0,0 +1,82 @@
+const mongoose = require("mongoose");
+
+
+const dayInformationSchema = {
+    active: {
+        type: Boolean,
+        default: 'true'
+    },
+    startTime: {
+        type: String,
+        required: this.active,
+        default: '00:00',
+        validate: [startTimeValidator]
+    },
+    endTime: {
+        type: String,
+        required: this.active,
+        default: '00:00',
+        validate: [endTimeValidator]
+    }
+}
+
+const availabilitySchema = new mongoose.Schema({
+    petSitterId: {
+        type: mongoose.Schema.Types.ObjectId,
+        required: true,
+        ref: 'Profile'
+    },
+    name: {
+        type: 'string',
+        required: true
+    },
+    days: {
+        sunday: dayInformationSchema,
+        monday: dayInformationSchema,
+        tuesday: dayInformationSchema,
+        wednesday: dayInformationSchema,
+        thursday: dayInformationSchema,
+        friday: dayInformationSchema,
+        saturday: dayInformationSchema
+    },
+
+});
+availabilitySchema.pre('validate', function (next) {
+    let startHour;
+    let endHour;
+    Object.entries(this.days).map(([key, value]) => {
+        Object.entries(value).map(([key, value]) => {
+
+            if (key == 'startTime') {
+                startHour = value.split(":")[0];
+            }
+            if (key == 'endTime') {
+                endHour = value.split(":")[0];
+            }
+            if (startHour > endHour) {
```

Comment:
Are we performing this validation on the FE too? If not, we could add it in :)

Reply:
Yes there is a validation for start time and end time on onChange event